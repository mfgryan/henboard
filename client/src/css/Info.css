.modal-body{
    max-height: 400px;
    overflow-y: auto;
}

.InfoItem {
    overflow-wrap: break-word;
}

.InfoItemClose {
    margin-top: 20px;
}
