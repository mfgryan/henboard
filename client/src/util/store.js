// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate } from "../util/middleware.js";
import data from "../util/data.js";
import rootReducer from "../reducers/index.js";

const store = createStore(rootReducer,applyMiddleware(logger,validate));

export { store }
