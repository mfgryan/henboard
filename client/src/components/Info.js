// react dep
import React from "react";
import { Modal, Col, Row, Button } from "react-bootstrap";

// component dep
import AddItemInfo from "./containers/AddItemInfo.js";

// utilities
import { url } from "../util/util.js";

// style
import "../css/Info.css";

const Info = ({ project, info, title, removeInfo, closeInfo }) => {
    return (
        <Modal.Dialog draggable="false">
            <Modal.Header>
                <AddItemInfo project={project} title={title} />
            </Modal.Header>
            {info.length > 0 &&
                <Modal.Body>
                    {info.map((info, index) =>
                        <Row key={index}>
                            <Col md={10}>
                                {!url().isUrl(info.value)
                                    ? info.value
                                    : <a href={url().checkProtocol(info.value)}>
                                          {info.value}
                                      </a>}
                            </Col>
                            <Col md={2}>
                                <p onClick={() => removeInfo(info)}>X</p>
                            </Col>
                        </Row>
                    )}
                </Modal.Body>}
            <Modal.Footer>
                <Button onClick={event => closeInfo(event, title)}>
                    Close
                </Button>
            </Modal.Footer>
        </Modal.Dialog>
    );
};

export default Info;
