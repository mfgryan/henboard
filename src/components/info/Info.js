// react dep
import React from "react";

// bootstrap dep
import { Modal, Col, Row, Button } from "react-bootstrap";

// url dep
import { urlUtil } from "../../util/Util";

const Info = ( { info, item, addInfo, removeInfo, changeItemValue, closeInfo } ) => {
    return (
        <Modal.Dialog draggable="false">
            <Modal.Header>
                <h2>{item.name}</h2>
                <input type="text" onChange={(event) =>changeItemValue(event,item)} value={item.value}/>
                <button onClick={() => addInfo(item)}>Add</button>
            </Modal.Header>
            {info.length > 0 && <Modal.Body>
                {info.map((info,index) =>
                <Row key={index} >
                    <Col md={10}>
                        {!urlUtil().isUrl(info.value) ? info.value :
                        <a href={urlUtil().checkProtocol(info.value)}>{info.value}</a>
                        }
                    </Col>
                    <Col md={2}>
                        <p onClick={() => removeInfo(info)}>X</p>
                    </Col>
                </Row>
                )}
            </Modal.Body>}
            <Modal.Footer>
                <Button onClick={(event) => closeInfo(event,item)}>Close</Button>
            </Modal.Footer>
        </Modal.Dialog>
    );
};

export default Info; 
