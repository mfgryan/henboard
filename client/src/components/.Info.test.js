import React from "react";
import Info from "./Info";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import { store } from "../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it('Info renders correctly', () => {
    let info = [{
        project: "henboard", 
        name: "test", 
        value: "hello"
    }];
    const tree = renderer.create(
        <Router>
            <Provider store={store}>
                <Info project="henboard" info={info} title="test" removeInfo={() => {}} closeInfo={() => {}}/>
            </Provider>
        </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
