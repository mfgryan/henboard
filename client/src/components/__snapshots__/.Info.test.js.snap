exports[`test Info renders correctly 1`] = `
<div
  className="modal"
  draggable="false"
  role="dialog"
  style={
    Object {
      "display": "block",
    }
  }
  tabIndex="-1">
  <div
    className="modal-dialog">
    <div
      className="modal-content"
      role="document">
      <div
        className="modal-header">
        <div
          className="AddItem">
          <h2
            onClick={[Function]}>
            test
          </h2>
          <div
            className="AddItemInput"
            style={
              Object {
                "visibility": "hidden",
              }
            }>
            <div
              className="form-group">
              <span
                className="input-group input-group-sm">
                <input
                  className="form-control"
                  id={undefined}
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  type="text"
                  value="" />
                <span
                  className="input-group-btn"
                  type="text">
                  <button
                    className="btn btn-default"
                    disabled={false}
                    onClick={[Function]}
                    type="button">
                    Add
                  </button>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="modal-body">
        <div
          className="row">
          <div
            className="col-md-10">
            hello
          </div>
          <div
            className="col-md-2">
            <p
              onClick={[Function]}>
              X
            </p>
          </div>
        </div>
      </div>
      <div
        className="modal-footer">
        <div
          className="dropdown btn-group">
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-default"
            disabled={false}
            id="Selector"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button">
            Send To
          </button>
          <ul
            aria-labelledby="Selector"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                2
                  
                <small>
                  05/15/17
                </small>
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                1
                  
                <small>
                  Backlog
                </small>
              </a>
            </li>
            <li
              onKeyDown={[Function]}
              onSelect={[Function]}>
              <a
                href="#"
                onClick={[Function]}>
                <b>
                  load all
                </b>
              </a>
            </li>
            <li
              className="divider"
              onKeyDown={[Function]}
              role="separator"
              style={undefined} />
          </ul>
        </div>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button">
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
